### ISSUE TITLE
Bootstrap lxc debian stretch fails to set apt-key

### ISSUE THREAD

**Issue by pauls1024** [saltstack/salt-bootstrap#1349]
### Description of Issue/Question
I had a working salt-master using salt-cloud to provision Debian Stretch based lxc containers on a Debian Stretch host.  Yesterday when I tried to build a new lxc container the bootstrap failed.  The LXC was built and started but the minion did not connect back to the master.

Upon logging in to the container I found that the repo had been added for the salt packages, but the apt-key had not run properly.  I think the bootstrap.sh had failed to get the GPG key from the salt repo.

I then downloaded a February copy of the bootstrap-salt.sh to /tmp on the host.  Renamed it to /tmp/custom-bootstrap.sh. Then edited the profile on the salt master to use /tmp/custom-bootstrap.sh.  Now with this workaround I can make and bootstrap lxc containers again properly.

### Setup
Debian stretch on master and minions
salt-cloud using LXC provider

### Steps to Reproduce Issue
Working config last week fails to boostrap this week as the salt repo apt-key is not installed.

### Versions and Systems
(`salt --versions-report`, `bootstrap-salt.sh -v`, system type and version,
cloud/VM provider as appropriate.)
Broken Version 
__ScriptVersion="2019.05.20"
__ScriptName="bootstrap-salt.sh"

Working Version
__ScriptVersion="2019.02.27"
__ScriptName="bootstrap-salt.sh"

Salt master versions report
root@salt:/etc/salt/cloud.deploy.d# salt --versions-report
Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Sep 26 2018, 18:42:22)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.9
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-5-amd64
         system: Linux
        version: debian 9.9

Salt minion (on LXC host) versions report

**Comment by lordfolken** [comment]
The problem is that the debian stretch images from linuxcontainer.org do not have one of the tools installed needed to fetch the key. The script tries: curl,wget and fetch. None of these are present in the images.
I tried to add these with the extra packages switch `-p curl`, however that is executed after the debian dependencies, and is therefore never executed since that part of the code is executed after the failing step.

I then copied the script to my own url: http://sonne.alt-f4.ch/~folken/bootstrap-salt.sh and added curl to the list of packages needed.
